var/spack/repos/builtin/packages/gapbs/package.py

```diff
@@ -0,0 +1,63 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Gapbs(MakefilePackage):
+    """The GAP Benchmark Suite is intended to help graph processing research by
+    standardizing evaluations. Fewer differences between graph processing
+    evaluations will make it easier to compare different research efforts and
+    quantify improvements. The benchmark not only specifies graph kernels,
+    input graphs, and evaluation methodologies, but it also provides an
+    optimized baseline implementation (this repo). These baseline
+    implementations are representative of state-of-the-art performance, and
+    thus new contributions should outperform them to demonstrate an
+    improvement."""
+
+    homepage = "http://gap.cs.berkeley.edu/benchmark.html"
+    url      = "https://github.com/sbeamer/gapbs/archive/v1.0.tar.gz"
+
+    version('1.0', 'ac2efa793f44e58553449f42b9779f3ff2d47634')
+
+    variant('serial', default=False, description='Version with no parallelism')
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter('Makefile')
+        makefile.filter(r'-std=c\+\+11', self.compiler.cxx11_flag)
+        makefile.filter('PAR_FLAG =.*',
+                        'PAR_FLAG=' + self.compiler.openmp_flag)
```

Comment:
Could you just specify `PAR_FLAG` and`CXX_FLAGS` as make target instead of editing the makefile?

Reply:
By make target, do you mean calling make like `make('PAR_FLAG=' + self.compiler.openmp_flag`)?

I'm not sure that would work, as in the [Makefile](https://github.com/sbeamer/gapbs/blob/master/Makefile), `PAR_FLAG` is unconditionally written to and the `-std=c++11` flag is hard coded.

Reply:
```
$ cat Makefile 
A=A
all:
	echo $(A)
$ make A=B
echo B
B
```